operator/watchers/cec.go

```diff
@@ -0,0 +1,38 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package watchers
+
+import (
+	"context"
+
+	"github.com/cilium/cilium/operator/pkg/ciliumenvoyconfig"
+	k8sClient "github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+// StartCECController starts the service watcher if it hasn't already and looks
+// for service of type with envoy enabled LB annotation. Once such service is
+// found, it will try to create one CEC associated with the service.
+func StartCECController(ctx context.Context, cfg ServiceSyncConfiguration, clientset k8sClient.Clientset) {
+	optsModifier, err := utils.GetServiceListOptionsModifier(cfg)
+	if err != nil {
+		log.WithError(err).Fatal("Error creating service option modifier")
+	}
+
+	swgSvcs := lock.NewStoppableWaitGroup()
+	swgEps := lock.NewStoppableWaitGroup()
+	InitServiceWatcher(cfg, clientset, swgSvcs, swgEps, optsModifier)
```

Comment:
Given that the `serviceInitOnce` prevents multiple informer instantiations, the `optsModifier` here won't have an effect because `StartCECController` will probably be called after the service informer is first initialized. Is that a problem?

Reply:
this is no longer applicable as per that latest changes on cached resource (e.g. legacy.resource.Services)